### ISSUE TITLE
Vega crashes with legend with empty data

### ISSUE THREAD

**Issue by domoritz** [vega/vega#1415]
When I start with this spec and insert data later, Vega freezes and never shows a chart even after I insert the data.

```json
{
  "$schema": "https://vega.github.io/schema/vega/v4.json",
  "data": [{"name": "data"}],
  "scales": [
    {
      "name": "color",
      "type": "ordinal",
      "domain": {"data": "data", "field": "foo"},
      "range": "category"
    }
  ],
  "legends": [{"fill": "color"}]
}
```

The canvas renderer crashes and the SVG renderer shows the error below. 

<img width="793" alt="screen shot 2018-09-08 at 18 14 26" src="https://user-images.githubusercontent.com/589034/45260118-0af20e00-b393-11e8-8de1-eb4417257026.png">

This looks similar to the problems in https://github.com/vega/vega/issues/1401. 

cc @arvind

**Comment by domoritz** [comment]
As a workaround, I hard code the domain, which resolves the issue for me.

**Comment by jheer** [comment]
Fixed in vega version 4.3.0.
